### PR TITLE
refactor: revamp `snapUtils` function - `dappUrl` 

### PR DESCRIPTION
**Pull request by stanleyyconsensys**
This PR is to change `snapUtils` function - `dappUrl` to using config value rather than env directly, to reduce the noice of process.env

This PR includes:
- move dappUrl to new utils file - `url.ts` and rename `dappUrl` to `getDappUrl`
- using enum rather than string to indicate  snap env, when it is not possible to pass in from client
- add snap env into `Config`.`snapEnv`  with default value

### CODE REVIEW

File: packages/starknet-snap/src/utils/url.ts
Comment (khanti42) [inline packages/starknet-snap/src/utils/url.ts:1]: Do we expect to have other url managed in url.ts ?
If it's only for the dapp why not calling it `dappUrl`, wdyt ?

**Review by khanti42 — COMMENTED** [review]
Shall we add an example in the env sample file ?

File: packages/starknet-snap/src/utils/url.ts
Comment (stanleyyconsensys) [reply packages/starknet-snap/src/utils/url.ts:1]: i think this is an helper for url, may be there will be some more url utils to come in, and it is very often case

Comment (stanleyyconsensys) [reply packages/starknet-snap/src/utils/url.ts:1]: what do u mean by example, if u are talking about "SNAP_ENV",  doesnt it already included in the env

**Review by khanti42 — DISMISSED** [review]
LGTM

**Review by khanti42 — APPROVED** [review]
LGTM

**Pull request merged**